packages/lesswrong/components/votes/lwReactions/DetailedReactionOverview.tsx

```diff
@@ -0,0 +1,191 @@
+import type { VotingProps } from '../votingProps';
+import { useNamesAttachedReactionsVoting } from './NamesAttachedReactionsVoteOnComment';
+import sumBy from 'lodash/sumBy';
+import { defineStyles, useStyles } from '../../hooks/useStyles';
+import ReactionIcon from "../ReactionIcon";
+import ReactOrAntireactVote from "./ReactOrAntireactVote";
+import LWTooltip from "../../common/LWTooltip";
+import { slugify } from '@/lib/utils/slugify';
+import Loading from '@/components/vulcan-core/Loading';
+
+const styles = defineStyles("DetailedReactionOverview", (theme: ThemeType) => ({
+  root: {
+    fontFamily: theme.typography.commentStyle.fontFamily,
+    maxWidth: 450, 
+    maxHeight: 400,
+    overflowY: 'auto',
+  },
+  itemRowContainer: { 
+    padding: '6px 0',
+    borderBottom: `1px solid ${theme.palette.border.faint}`,
+    '&:last-child': {
+      borderBottom: 'none',
+    }
+  },
+  itemRow: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+  },
+  iconAndLabel: {
+    display: 'flex',
+    alignItems: 'center',
+    flexGrow: 1, 
+    marginRight: 8,
+    minWidth: 0, 
+    paddingLeft: 4,
+  },
+  iconAndLabelTextWrapper: { 
+    display: 'flex',
+    flexDirection: 'column',
+    overflow: 'hidden',
+    marginLeft: 8,
+  },
+  labelText: {
+    fontWeight: 'bold',
+    display: '-webkit-box',
+    '-webkit-line-clamp': 2,
+    '-webkit-box-orient': 'vertical',
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    color: theme.palette.text.primary,
+  },
+  quoteTextDisplay: {
+    fontSize: '0.9em',
+    color: theme.palette.text.primary,
+    display: '-webkit-box',
+    '-webkit-line-clamp': 2,
+    '-webkit-box-orient': 'vertical',
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+  },
+  voteButtonsWrapper: { 
+    flexShrink: 0,    
+  },
+  usersWhoReactedList: {
+    fontSize: '12px',
+    color: theme.palette.grey[500],
+    paddingLeft: 30,
+    marginTop: 4,
+  }
+}));
+
+interface DisplayableReactionItem {
+  id: string;
+  reactionName: EmojiReactName;
+  reactionLabel: string;
+  iconComponent: React.ReactNode;
+  quote: QuoteLocator | null;
+  netCount: number;
+  currentUserVote: VoteOnReactionType | null;
+  reactions: UserReactInfo[];
+}
+
+interface DetailedReactionOverviewProps {
+  voteProps: VotingProps<VoteableTypeClient>;
+}
+
+const DetailedReactionOverview = ({ voteProps }: DetailedReactionOverviewProps) => {
+  const classes = useStyles(styles);
+  const { getCurrentUserReactionVote, setCurrentUserReaction } = useNamesAttachedReactionsVoting(voteProps);
+  const UsersWhoReacted = React.lazy(() => import('./UsersWhoReacted'));
+
+  const normalizedReactionsList = getNormalizedReactionsListFromVoteProps(voteProps);
+  const allUsedReactsMap: NamesAttachedReactionsList = normalizedReactionsList?.reacts ?? {};
+
+  const displayItems: DisplayableReactionItem[] = [];
+
+  masterReactionList.forEach(reactionDetail => {
```

Comment:
I kinda suspect you don't need to be mapping over `namesAttachedReactions` rather than the keys of `allUsedReactsMap` - is it just to get the `name`?

Separately I'd prefer it if this were more functional (rather than pushing into displayItems from inside of a forEach), and kinda suspect the current implementation is prone to a bunch of rerenders.